data/json/mapgen_palettes/shelter.json

```diff
@@ -27,13 +28,156 @@
+      "!": "t_pavement",
+      "`": "t_pavement_y",
+      "&": "t_sidewalk",
+      "#": "t_soil",
+      "+": [ "t_door_c", "t_door_b" ],
+      "-": "t_wall_w",
+      "_": "t_linoleum_white",
+      ":": [ "t_window_frame", "t_window_no_curtains", "t_window" ],
+      ">": "t_stairs_down",
+      "<": "t_stairs_up",
+      "|": "t_wall_w",
+      ";": "t_concrete_wall",
+      "/": "t_concrete_wall",
+      "*": "t_ladder_up",
+      "=": [ "t_door_b", "t_door_locked_interior", "t_door_c", "t_door_o" ],
+      "4": "t_gutter_downspout"
+    },
+    "furniture": {
+      "6": "f_console",
+      "x": "f_console",
+      "b": "f_bench",
+      "c": "f_cupboard",
+      "l": [ "f_locker", "f_wreckage" ],
+      "S": "f_sink",
+      "%": "f_trashcan",
+      "U": [ "f_utility_shelf", "f_wreckage" ],
+      "B": [ "f_utility_shelf", "f_wreckage" ],
+      "t": "f_table",
+      "C": "f_counter"
+    },
+    "toilets": { "T": {  } },
+    "items": {
+      "l": { "item": "shelter_supplies", "chance": 40 },
+      "c": [ { "item": "trash", "chance": 1 }, { "item": "softdrugs", "chance": 2 }, { "item": "shelter_supplies", "chance": 10 } ],
+      "b": [ { "item": "softdrugs", "chance": 2 }, { "item": "shelter_supplies", "chance": 1 }, { "item": "trash", "chance": 1 } ],
+      ".": [ { "item": "trash", "chance": 1 }, { "item": "trash_forest", "chance": 1 } ],
+      "%": [ { "item": "trash", "chance": 20, "repeat": [ 1, 3 ] } ],
+      "S": [ { "item": "softdrugs", "chance": 2 }, { "item": "shelter_supplies", "chance": 10 }, { "item": "trash", "chance": 1 } ],
+      "U": [ { "item": "softdrugs", "chance": 2 }, { "item": "shelter_supplies", "chance": 10 }, { "item": "trash", "chance": 1 } ],
+      "B": [ { "item": "softdrugs", "chance": 2 }, { "item": "shelter_supplies", "chance": 10 }, { "item": "trash", "chance": 1 } ],
+      "t": [ { "item": "softdrugs", "chance": 2 }, { "item": "shelter_supplies", "chance": 10 }, { "item": "trash", "chance": 1 } ],
+      "C": [ { "item": "softdrugs", "chance": 2 }, { "item": "shelter_supplies", "chance": 10 }, { "item": "trash", "chance": 1 } ],
+      "_": [ { "item": "trash", "chance": 1 } ]
+    },
+    "nested": {
+      "|": { "chunks": [ [ "shelter_graffiti", 5 ], [ "general_graffiti", 20 ], [ "null", 75 ] ] },
+      "-": { "chunks": [ [ "shelter_graffiti", 5 ], [ "general_graffiti", 20 ], [ "null", 75 ] ] },
+      ";": { "chunks": [ [ "shelter_graffiti", 5 ], [ "general_graffiti", 20 ], [ "null", 75 ] ] },
+      "/": { "chunks": [ [ "shelter_graffiti", 5 ], [ "general_graffiti", 20 ], [ "null", 75 ] ] },
+      "6": { "chunks": [ [ "shelter_computer", 50 ], [ "shelter_computer_broken", 50 ] ] },
+      "x": { "chunks": [ [ "shelter_computer", 20 ], [ "shelter_computer_broken", 80 ] ] }
+    }
+  },
+  {
+    "type": "palette",
+    "id": "shelter_zombies",
+    "monsters": { "_": { "monster": "GROUP_ZOMBIE", "density": 0.0005 }, "b": { "monster": "GROUP_ZOMBIE", "density": 0.0001 } }
```

Comment:
I would be very careful with the density parameter here and verify what it does. If it does the same thing as it does in `place_monster` and `place_monsters`, then this flag will multiply with the world spawn rate and round to the nearest integer.

Reply:
Frankly, I would advise against using this JSON structure to place monsters, as it's difficult to control. Suggest you use the `place_monster` flag instead, as it lets you determine exactly how many times to roll the dice in a clear way.

Reply:
It seems to be placing a handful of zombies, which is what I intended. I'm trying to avoid having this be a completely crowded location, since it basically never spawned anything before. My assumption is that only a few people made it to these shelters and they then turned after the cataclysm.

Reply:
If you've run tests and things are working as intended, that's all well and good then. I just didn't want a situation like we had with hospitals and parks, where there was an absurd number of zombies.

Reply:
It's working well enough for me at least. The screenshots only show a couple zombies in them. I've only ever seen about 5 zombies spawn from it.